github-notifications/main.py

```diff
@@ -0,0 +1,111 @@
+
+    if "ref_type" in webhook_payload:
+        if webhook_payload["ref_type"] == "branch":
+            return_status = create_branch(webhook_payload)
+    
+    # if webhook_payload_list[1] == "ref_type":
+    # elif webhook_payload_list[2] == "pull_request":
+    if "pull_request" in webhook_payload:
+        return_status = pull_request(webhook_payload)
+    else:
+        print("Key doesn't exist in JSON data")
+        
+    if return_status == None:
+        return_status = "Incomplete bc Null"
+    return return_status
+
+def create_branch(payload):
+    description = get_description(payload, "repository", "description")
+    sender = payload["sender"]["login"]
+    repository = payload["repository"]["full_name"]
+    url = payload["repository"]["html_url"] + "/branches"
+    text = ":tree-branch: Branch created"
+    return_status = write_message(text,"Branch ", url, repository, description, sender)
+    return return_status
+
+def pull_request(payload):
+    sender = payload["sender"]["login"]
+    pr_url = payload["pull_request"]['html_url']
+    repository = payload["repository"]["full_name"]
+    text = ":pusheencomputer: Pull Request " + payload["action"]
+    repository = payload["repository"]["full_name"]
+    description = get_description(payload, "pull_request", "title")
+    return_status = write_message(text,"Pull Request", pr_url, repository, description, sender)
+    return return_status
+
+def get_description(payload, index1, index2):
+    description = " "
+    if payload[index1][index2] != None:
+        description = payload[index1][index2]
+    return description
+
+def write_message(text, type, url, repo, description, user): 
+    # slack_token = os.environ["SLACK_API_TOKEN"]
+    slack_token = os.getenv('SLACK_API_TOKEN')
+    client = WebClient(token=slack_token)
+    return_status = "Incomplete"
+    
+    try:
+        response = client.chat_postMessage(
+                channel="#tech-test",
+                blocks=[
+                {
+                    "type": "section",
+                    "text": {
+                        "type": "mrkdwn",
+                        "text": text + " on " + repo + "."
+                    },
+                    "accessory": {
+                        "type": "button",
+                        "text": {
+                            "type": "plain_text",
+                            "text": "View " + type + "!"
+                            # "emoji": true
+                        },
+                        "value": "click_me_123",
+                        "url": url,
+                        "action_id": "button-action"
+                    }
+                },
+                {
+                    "type": "context",
+                    "elements": [
+                        {
+                            "type": "mrkdwn",
+                            "text": description + " - " + user
+                        }
+                    ]
+                }
```

Comment:
In the event that these happen one after another, I'd recommend a visual separator here. A divider might do the trick. [What do you think](https://app.slack.com/block-kit-builder/T0252VA3B#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Pull%20Request%20opened%20on%20texastribune/texastribune.%22%7D,%22accessory%22:%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22View%20Pull%20Request%22%7D,%22value%22:%22click_me_123%22,%22url%22:%22http://foo.com%22,%22action_id%22:%22button-action%22%7D%7D,%7B%22type%22:%22context%22,%22elements%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22Bring%20texastribune/tt-docker-base%20back%20into%20this%20repo%20:whale:%20-%20bfreeds%22%7D%5D%7D,%7B%22type%22:%22divider%22%7D%5D%7D)?
<img width="486" alt="Screen Shot 2022-12-02 at 5 15 13 PM" src="https://user-images.githubusercontent.com/2974713/205404967-3e11422a-9007-4b66-888b-8d7638f8a9cf.png">